src/ast/codegen_llvm.cpp

```diff
@@ -1647,28 +1640,26 @@ void CodegenLLVM::visit(AssignMapStatement &assignment)
   if (!expr_) // Some functions do the assignments themselves
     return;
 
-  Value *val, *expr;
-  expr = expr_;
-  AllocaInst *key = getMapKey(map);
```

Comment:
What is the reason for moving this mapkey down? It seems to cause a lot of codegen changes making it harder to review. Like tests/codegen/llvm/args_multiple_tracepoints_category_wild.ll  seems to only a order change, not a functional one

Reply:
yup, it's just an order change. I wanted to make the lifetime of the map key as short as possible. should I delete it? it's only a nice-to-have.

I think either way I expect a lot of test churn, because the scoped_deleter emits its lifetime-end in a later location than when we did it manually. should I try to control where it emits its lifetime-end, to further reduce test churn?

Reply:
> yup, it's just an order change. I wanted to make the lifetime of the map key as short as possible. should I delete it? it's only a nice-to-have.

If it has a significant impact on the amoun t  of changes in this PR it might be better to split it to a separate pr (or commit) to make reviewing easier.

`scripts/compare_tool_elf.sh` might be useful here too, just to make sure existing tools aren't impacted in a bad way.

> I think either way I expect a lot of test churn, because the scoped_deleter emits its lifetime-end in a later location than when we did it manually. should I try to control where it emits its lifetime-end, to further reduce test churn?

Don't think thats worth the effort.